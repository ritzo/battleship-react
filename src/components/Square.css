.square {
  background: #fff;
  border: 1px solid #999;
  float: left;
  font-size: 24px;
  font-weight: bold;
  line-height: 34px;
  height: 34px;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 34px;
}

.square:focus {
  outline: none;
}

.empty {
  background: hsl(234, 90%, 45%);
}

.invalid {
  background: hsl(234, 90%, 45%);
  cursor: not-allowed;
}

.ship {
  background: rgb(6, 75, 35);
}

.shoot_miss {
  background: rgb(86, 153, 240);
}

.shoot_ship {
  background: rgb(88, 6, 6);
}
