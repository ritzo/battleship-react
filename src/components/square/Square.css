.square {
  background: #fff;
  border: 1px solid #999;
  float: left;
  font-size: 24px;
  font-weight: bold;
  line-height: 34px;
  height: 34px;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 34px;
}

.square:focus {
  outline: none;
}

.empty {
  background: hsl(234, 85%, 61%);
}

.invalid {
  background: hsl(234, 85%, 61%);
}

.edition .invalid {
  background: hsl(233, 60%, 81%) !important;
  cursor: not-allowed;
}

.ship {
  background: rgb(6, 75, 35);
  border: solid 1px rgb(6, 75, 35);

}

.shoot_miss {
  background: rgb(86, 153, 240);
}

.shoot_ship {
  background: rgb(206, 104, 64);
  border: solid 1px rgb(206, 104, 64);
}

.ship_sunken {
  background: rgb(252, 3, 3);
  border: solid 1px rgb(252, 3, 3);
}

.last-hit{
  border: 5px solid rgba(9, 64, 78, 0.959);
}
