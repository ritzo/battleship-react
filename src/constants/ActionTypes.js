// Actions related to the game setup
export const SETUP_BOARD = 'SETUP_BOARD';
export const SET_NAME = 'SET_NAME';
export const ADD_SHIP = 'ADD_SHIP';
export const PLAY = 'PLAY';
// Actions related to the game itself
export const SURRENDER = 'SURRENDER';
export const SHOOT = 'SHOOT';
export const OPONENT_TURN = 'OPONENT_TURN';
