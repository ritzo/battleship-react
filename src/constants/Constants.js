export const STATES = {
  OPEN: 'open',
  PLAYING: 'playing',
  FINISHED: 'finished',
};

export const SHIPS = [2, 3, 3, 3, 4];

export const POSITIONS = {
  HORIZONTAL: 'horizontal',
  VERTICAL: 'vertical',
};

export const PLAYERS = {
  PLAYER: 'player',
  OPONENT: 'cpu',
};

export const SQUARE_STATES = {
  EMPTY: 'empty',
  INVALID: 'invalid',
  SHIP: 'ship',
  SHOOT_MISS: 'shoot_miss',
  SHOOT_SHIP: 'shoot_ship',
};

export const BOARD_TYPES = {
  EDITION: 'edition',
  PLAYER: 'player',
  OPONENT: 'oponent',
};
