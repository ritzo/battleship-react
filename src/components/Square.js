import React from 'react';
import PropTypes from 'prop-types';

import './Square.css';

class Square extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: props.value,
      onClick: props.onClick,
    };
  }

  render() {
    const { value, onClick } = this.state;
    const className = `square ${value}`;

    return (
      <button
        type="button"
        className={className}
        onClick={onClick}
      />
    );
  }
}

Square.propTypes = {
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Square;
