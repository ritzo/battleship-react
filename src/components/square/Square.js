import React from 'react';
import PropTypes from 'prop-types';
import { ORIENTATION } from '../../constants/Constants';

import './Square.css';

class Square extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      x: props.x,
      y: props.y,
    };
  }

  render() {
    const { x, y } = this.state;
    const { matrix } = this.props;
    const value = matrix[y][x];
    const className = `square ${value}`;

    this.onClickHandler = (event, index) => {
      const { actions } = this.props;

      let orientation = ORIENTATION.HORIZONTAL;
      if (event.type === 'contextmenu') {
        orientation = ORIENTATION.VERTICAL;
      }

      const clickParam = Object.assign({}, index, { orientation });

      if (actions.onClickHandler) {
        actions.onClickHandler(clickParam);
      }
      // Prevents the right button defaults action.
      event.preventDefault();
    };

    return (
      <button
        type="button"
        className={className}
        onClick={e => this.onClickHandler(e, { x, y })}
        onContextMenu={e => this.onClickHandler(e, { x, y })}
      />
    );
  }
}

Square.propTypes = {
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  matrix: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string.isRequired).isRequired).isRequired,
  actions: PropTypes.shape({
    onClickHandler: PropTypes.func,
  }).isRequired,
};

export default Square;
