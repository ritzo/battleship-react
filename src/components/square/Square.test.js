import React from 'react';
import { shallow } from 'enzyme';

import Square from './Square';

import { SQUARE_STATES } from '../../constants/Constants';

describe('Square', () => {
  it('Snapshot', () => {
    const x = 5;
    const y = 1;
    const matrix = Array(10).fill([]).map((() => Array(10).fill(SQUARE_STATES.EMPTY)));
    const actions = {
      onClickHandler() {},
    };

    const enzymeWrapper = shallow(
      <Square x={x} y={y} matrix={matrix} actions={actions} />,
    );

    expect(enzymeWrapper).toMatchSnapshot();
  });
});
