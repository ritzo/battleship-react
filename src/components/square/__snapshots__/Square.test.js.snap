// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Snapshot 1`] = `
<button
  className="square empty"
  onClick={[Function]}
  onContextMenu={[Function]}
  type="button"
/>
`;
